### ISSUE TITLE
Shared configuration for services

### ISSUE THREAD

**Issue by atrauzzi** [rancher/rancher#3798]
If I'm running multiple containers - services - that require the same environment variables (and possibly also configuration like volumes, etc), it might be nice if I could specify one central profile for them to all read from.

This isn't so much templates, as I'd like to be able to change the configuration and optionally "upgrade" all related containers with the new config.

As is expected, this system would cascade such that if my shared configuration specifies one thing, but a specific container overrides it, the most specific wins out.

**Comment by will-chan** [comment]
We are a feature called meta-data service that you can use to pass in data and have it be accessible by containers within the same service or even across services.  Would this work for you?

http://docs.rancher.com/rancher/metadata-service/

**Comment by atrauzzi** [comment]
No, what I'm talking about is configuration at a container level, not application.

This would be too much overhead for me to get the values on every request (PHP).

Also, I don't want to extend every container I use to query a Rancher specific metadata service.

**Comment by atrauzzi** [comment]
Also, I'd love to see templates of variables as an option as well.  If I'm running multiple services that all tend to need the same set of variables, being able to say "use this template" and have it barf a bunch of un-set varaibles into the service configuration would be great.
